### PR TITLE
Add terraform plugin into backstage-app

### PR DESCRIPTION
**Pull request by Enclavet**
Updated terraform plugin to include tfstates stored as k8s secrets
Refactored backend for new system
Refactored fetch component to move fetches to API
Updated backstage-app to include the terraform plugin by default

### CODE REVIEW

File: plugins/terraform/package.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    "postpack": "backstage-cli package postpack"
+  },
+  "dependencies": {
+    "@backstage/core-components": "~0.13.10",
+    "@backstage/core-plugin-api": "~1.8.2",
+    "@backstage/theme": "~0.5.0",
+    "@material-ui/core": "^4.9.13",
+    "@material-ui/icons": "^4.9.1",
+    "@material-ui/lab": "^4.0.0-alpha.61",
+    "react-use": "^17.2.4"
```
Comment (nabuskey) [inline plugins/terraform/package.json:33]: Seems out of sync with the rest of the packages. Try running `yarn backstage-cli versions:check` It should tell you what to change.

File: plugins/terraform/package.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  },
+  "devDependencies": {
+    "@backstage/cli": "~0.25.1",
+    "@backstage/core-app-api": "~1.11.3",
+    "@backstage/dev-utils": "^1.0.26",
+    "@backstage/test-utils": "^1.4.7",
+    "@testing-library/jest-dom": "^5.10.1",
+    "@testing-library/react": "^12.1.3",
+    "@testing-library/user-event": "^14.0.0",
+    "msw": "^1.0.0"
```
Comment (nabuskey) [inline plugins/terraform/package.json:46]: same thing here. `yarn backstage-cli versions:check`

File: plugins/terraformv2-backend/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# terraformv2
+
+Welcome to the terraformv2 backend plugin!
```
Comment (nabuskey) [inline plugins/terraformv2-backend/README.md:3]: I think we can just name it terraform-backend?  Applies to the name of this folder too and the package name too

**Review by nabuskey — COMMENTED** [review]
We need to keep the dependencies in sync. I've had some terrible experiences when I didn't do that before.

File: plugins/terraformv2-backend/README.md (hunk 3 above)
Comment (andklee) [reply plugins/terraformv2-backend/README.md]: Renamed it.

File: plugins/terraform/package.json (hunk 2 above)
Comment (andklee) [reply plugins/terraform/package.json]: Updated.

File: plugins/terraform/package.json (hunk 1 above)
Comment (andklee) [reply plugins/terraform/package.json]: Updated.

File: plugins/terraform/src/api/Terraform.test.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("can fetch from k8s", async () => {
+    mockKClient.proxy.mockResolvedValue({
+      status: 200,
+      ok: true,
+      text: async () => "teststring",
```
Comment (nabuskey) [inline plugins/terraform/src/api/Terraform.test.ts:36]: LGTM

**Review by nabuskey — APPROVED** [review]

**Pull request merged**